exchange/message/message.go

```diff
@@ -0,0 +1,47 @@
+package message
+
+import (
+	"context"
+	"errors"
+	"fmt"
+)
+
+var errChannelBufferFull = errors.New("channel buffer is full")
+
+// Relay defines a channel relay for messages
+type Relay struct {
+	C    <-chan Payload
```

Comment:
🥃 🗻  Just a thought: 
This feels like we're introducing a duplicate of `websocketDataReceiver` 's fan out to datahandlers pattern.
I found myself wanting to say "Okay. I've subscribed to X. Now I just want to get that data" a few times. And I have to Register a Datahandler and discard what I'm not interested in.
Worse, I'm subscribing to all exchanges when I do that, so I'm filtering a lot.

It'd be good if we could merge that into here, and allow targetted relaying.

This is out of scope, though